nmdc_schema/migrators/migrator_from_9_3_to_9_4.py

```diff
@@ -16,16 +15,24 @@ def __init__(self, *args, **kwargs) -> None:
 
     def move_extraction_qc_status(self, extraction: dict) -> dict:
         r"""
-        Move quality_control_report.status to qc_status
+        Updates the `Extraction` so that the value originally in its `quality_control_report.status` field, if any,
+        is stored in a new field named `qc_status`; and the `quality_control_report` field no longer exists.
 
         >>> m = Migrator()
         >>> m.move_extraction_qc_status({'id': 123, 'quality_control_report': {'status': 'pass'}})
         {'id': 123, 'qc_status': 'pass'}
+        >>> m.move_extraction_qc_status({'id': 123, 'quality_control_report': {'status': 'pass', 'name': 'n'}})
+        {'id': 123, 'qc_status': 'pass'}
+        >>> m.move_extraction_qc_status({'id': 123, 'quality_control_report': {'name': 'n'}})
```

Comment:
we dont have any records like this currently but it is possible to have quality_control_report:{'status': 'fail'} or 'quality_control_report': {'status': 'pass', 'name': 'n'}} so may want to add those cases as well in case the data in mongo changes b/c now and the prod migration.

Reply:
I'll add doctests for some specific cases, too. 

Note: These things are tests, so don't show every possible input value; but I do intend for them to show enough to test each code branch and also illustrate the behavior of the function.

Reply:
I have added the following doctests in a new commit:

```py
        >>> m.move_extraction_qc_status({'id': 123, 'quality_control_report': {'status': 'fail'}})
        {'id': 123, 'qc_status': 'fail'}
```

```py
        >>> m.move_extraction_qc_status({'id': 123, 'quality_control_report': {'status': 'fail', 'name': 'n'}})
        {'id': 123, 'qc_status': 'fail'}
```

Reply:
> ...or `'quality_control_report': {'status': 'pass', 'name': 'n'}}`

I assumed you meant `'status': 'fail'` there.